### PR TITLE
Fix translation of relational operations

### PR DESCRIPTION
**Pull request by Fznamznon**
Whereas it is not possible to generate vector of bools in FE,
we have to change return type for corresponding instructions in SPIRV
translator to vector of bools. SPIRV translator already did this for
some instructions, this patch extends this behaviour to handle more
instructions.

### CODE REVIEW

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -2417,24 +2417,28 @@ LLVMToSPIRV::transBuiltinToInstWithoutDecoration(Op OC, CallInst *CI,
...
       Type *BoolTy = IntegerType::getInt1Ty(M->getContext());
       auto IsVector = ResultTy->isVectorTy();
       if (IsVector)
         BoolTy = VectorType::get(BoolTy, ResultTy->getVectorNumElements());
       auto BBT = transType(BoolTy);
-      auto Cmp = BM->addCmpInst(OC, BBT, transValue(CI->getArgOperand(0), BB),
-                                transValue(CI->getArgOperand(1), BB), BB);
+      SPIRVInstruction *Res;
+      if (isCmpOpCode(OC))
+        Res = BM->addCmpInst(OC, BBT, transValue(CI->getArgOperand(0), BB),
```
Comment (svenvh) [inline lib/SPIRV/SPIRVWriter.cpp]: You could move the assert that you removed above into this `if` (and also add an assert `CI && CI->getNumArgOperands() == 1` in the `else`.

Comment (Fznamznon) [reply lib/SPIRV/SPIRVWriter.cpp]: Sure, added.

File: lib/SPIRV/libSPIRV/SPIRVOpCode.h [hunk 2]
```diff
@@ -74,6 +74,10 @@ inline bool isLogicalOpCode(Op OpCode) {
   return (unsigned)OpCode >= OpLogicalEqual && (unsigned)OpCode <= OpLogicalNot;
 }
 
+inline bool isRelationalOpCode(Op OpCode) {
```
Comment (AlexeySotkin) [inline lib/SPIRV/libSPIRV/SPIRVOpCode.h]: What do you think about `isUnaryPredicateOpCode` name for the function?

Comment (Fznamznon) [reply lib/SPIRV/libSPIRV/SPIRVOpCode.h]: Hmm, okay.

Comment (Fznamznon) [reply lib/SPIRV/libSPIRV/SPIRVOpCode.h]: Done.

**Comment by AlexeySotkin** [comment]
@Fznamznon, could you squash your commits?

**Comment by bader** [comment]
> @Fznamznon, could you squash your commits?

I/you can squash during merge. Do you have any other comments?

**Comment by bader** [comment]
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/cb3be8626ec224aaf1bdf76a26ac5232b34d43fa - squashed commit.

**Pull request merged**